wcfsetup/install/files/ts/WoltLabSuite/Core/Ui/Dropdown/Simple.ts

```diff
@@ -0,0 +1,596 @@
+        }
+      }
+
+      const itemList = menu.querySelector('.scrollableDropdownMenu');
+      if (itemList !== null) {
+        itemList.classList[(itemList.scrollHeight > itemList.clientHeight ? 'add' : 'remove')]('forceScrollbar');
+      }
+
+      notifyCallbacks(containerId, 'open');
+
+      if (!disableAutoFocus) {
+        menu.setAttribute('role', 'menu');
+        menu.tabIndex = -1;
+        menu.removeEventListener('keydown', dropdownMenuKeyDown);
+        menu.addEventListener('keydown', dropdownMenuKeyDown);
+        menu.querySelectorAll('li').forEach(listItem => {
+          if (!listItem.clientHeight) return;
+          if (firstListItem === null) firstListItem = listItem;
+          else if (listItem.classList.contains('active')) firstListItem = listItem;
+
+          listItem.setAttribute('role', 'menuitem');
+          listItem.tabIndex = -1;
+        });
+      }
+
+      UiDropdownSimple.setAlignment(dropdown, menu, alternateElement);
+
+      if (firstListItem !== null) {
+        firstListItem.focus();
+      }
+    }
+  });
+
+  window.WCF.Dropdown.Interactive.Handler.closeAll();
+
+  return (event === null);
+}
+
+function handleKeyDown(event: KeyboardEvent): void {
+  // <input> elements are not valid targets for drop-down menus. However, some developers
+  // might still decide to combine them, in which case we try not to break things even more.
+  const target = event.currentTarget as HTMLElement;
+  if (target.nodeName === 'INPUT') {
+    return;
+  }
+
+  if (event.key === 'Enter' || event.key === 'Space') {
+    event.preventDefault();
+    toggle(event);
+  }
+}
+
+function dropdownMenuKeyDown(event: KeyboardEvent): void {
+  let button, dropdown;
+
+  const activeItem = document.activeElement as HTMLElement;
+  if (activeItem.nodeName !== 'LI') {
+    return;
+  }
+
+  if (event.key === 'ArrowDown' || event.key === 'ArrowUp' || event.key === 'End' || event.key === 'Home') {
+    event.preventDefault();
+
+    const listItems = Array.from<HTMLElement>(activeItem.closest('.dropdownMenu')!.querySelectorAll('li'));
+    if (event.key === 'ArrowUp' || event.key === 'End') {
+      listItems.reverse();
+    }
+
+    let newActiveItem: HTMLElement | null = null;
+    const isValidItem = listItem => {
+      return !listItem.classList.contains('dropdownDivider') && listItem.clientHeight > 0;
+    };
+
+    let activeIndex = listItems.indexOf(activeItem);
+    if (event.key === 'End' || event.key === 'Home') {
+      activeIndex = -1;
+    }
+
+    for (let i = activeIndex + 1; i < listItems.length; i++) {
+      if (isValidItem(listItems[i])) {
+        newActiveItem = listItems[i];
+        break;
+      }
+    }
+
+    if (newActiveItem === null) {
+      for (let i = 0; i < listItems.length; i++) {
+        if (isValidItem(listItems[i])) {
+          newActiveItem = listItems[i];
+          break;
+        }
+      }
```

Comment:
```suggestion
      newActiveItem = listItems.find(isValidItem);
```

see: https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/find